File: src/main/java/myToolWindow/SaveConfirmationDialog.java
package myToolWindow;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.awt.*;
import java.awt.event.*;

public class SaveConfirmationDialog extends DialogWrapper {
    private JPanel contentPane;
    private JTextField settingsName;

    public SaveConfirmationDialog(@Nullable final Project project) {
        super(project);
        setModal(true);
        setTitle("Save Settings");

        init();
    }


    @Nullable
    @Override
    protected JComponent createCenterPanel() {
        return contentPane;
    }

    public String getSettingsName() {
        return settingsName.getText();
    }

    public void setSettingsName(String settingsName) {
        this.settingsName.setText(settingsName);
    }

    @Nullable
    @Override
    public JComponent getPreferredFocusedComponent() {
        return settingsName;
    }
}

File: src/main/java/myToolWindow/SaveConfirmationDialog.kt
package myToolWindow

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import javax.swing.JComponent
import javax.swing.JPanel
import javax.swing.JTextField

class SaveConfirmationDialog(project: Project?) : DialogWrapper(project) {

    private var contentPane: JPanel? = null

    private var settingsName: JTextField? = null

    init {
        isModal = true
        title = "Save Settings"

        init()
    }

    override fun createCenterPanel(): JComponent? {
        return contentPane
    }

    fun getSettingsName(): String {
        return settingsName!!.text
    }

    fun setSettingsName(settingsName: String?) {
        this.settingsName!!.text = settingsName
    }

    override fun getPreferredFocusedComponent(): JComponent? {
        return settingsName
    }
}
